js/pro/poloniex.js

```diff
@@ -0,0 +1,886 @@
+            'channel': [ 'auth' ],
+            'params': {
+                'key': this.apiKey,
+                'signTimestamp': timestamp,
+                'signature': signature,
+                'signatureMethod': 'HmacSHA256',  // optional
+                'signatureVersion': '2',          // optional
+            },
+        };
+        return await this.watch (url, accessPath, request, accessPath);
+    }
+
+    async subscribe (name, publicOrPrivate, symbols = undefined, params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @description Connects to a websocket channel
+         * @param {String} name name of the channel
+         * @param {String} publicOrPrivate "public" or "private"
+         * @param {[String]|undefined} symbols CCXT market symbols
+         * @param {Object} params extra parameters specific to the poloniex api
+         * @returns {Object} data from the websocket stream
+         */
+        await this.loadMarkets ();
+        const url = this.urls['api']['ws'][publicOrPrivate];
+        // if ('signature' in params) {
+        //     // need to distinguish between public trades and user trades
+        //     url = url + '?';
+        // }
+        const subscribe = {
+            'event': 'subscribe',
+            'channel': [
+                name,
+            ],
+        };
+        const marketIds = [ ];
+        if (symbols !== undefined) {
+            if (symbols !== [ 'all' ]) {
+                for (let i = 0; i < symbols.length; i++) {
+                    const symbol = symbols[i];
+                    marketIds.push (this.marketId (symbol));
+                }
+            } else {
+                marketIds.push ('all');
+            }
+            subscribe['symbols'] = marketIds;
+        } else {
+            marketIds.push ('all');
+        }
+        const messageHash = name + ':' + marketIds.join (',');
+        const request = this.extend (subscribe, params);
+        return await this.watch (url, messageHash, request, name);
+    }
+
+    async watchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchOHLCV
+         * @description watches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @see https://docs.poloniex.com/#public-channels-market-data-candlesticks
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
+         */
```

Comment:
this object should live inside `options`.

Reply:
I don't know what you're trying to say

Reply:
I believe he means to move line 138 to 152 to options defined on L35.
However I see you already have an object in options called `channelToTimeframe` I would try to use that one to map the values. You can use the function `findTimeframe()` to help